### ISSUE TITLE
vagrant plugin install /path/to/your/bugfixed.gem fails

### ISSUE THREAD

**Issue by dmatora** [mitchellh/vagrant#5537]
I have a repo with my vagrant project and bugfixed gems (like vagrant-sshfs-0.0.7.gem vagrant-unison-0.0.13.gem)
i use to be able to install them, by copy pasting commands from readme.md i've created
```
vagrant plugin install vagrant-unison-0.0.15.gem
vagrant plugin install vagrant-sshfs-0.0.7.gem
```
now it just produces empty plugin folders
~\.vagrant.d\gems\gems\vagrant-sshfs-0.0.7
~\.vagrant.d\gems\gems\vagrant-unison-0.0.15
i tried vagrant 1.7.2 / 1.7.1 / 1.7.0 / 1.6.0 and it's all the same
it used to be working, and now it just does't (OSX 10.10.2)
any clue, why?

**Comment by sethvargo** [comment]
Hi @dmatora 

I am sorry you are having issues. Could you try specifying the full path to the `.gem` file instead of just the name?

It would be very helpful to know if you experience this issue with other third-party plugins. For example, can you clone the vagrant-aws project, build the gem, and try to install it? Do you see the same issue?

Finally, it would be very helpful to see the output of:

```
env | grep -i vagrant
```

to see which Vagrant environment variables you have. Please also try running the plugin install commands in debug mode.

**Comment by dmatora** [comment]
- full path did't help
- these plugins have successfully installed
  - vagrant-aws-0.6.0.gem
  - vagrant-parallels-1.3.9.gem
  - vagrant-unison-0.0.13.gem
- these plugins didn't
  - vagrant-digitalocean-0.7.3.gem
  - vagrant-host-shell-0.0.4.gem
  - vagrant-hostsupdater-0.0.11.gem
  - vagrant-nfs_guest-0.1.4.gem
  - vagrant-sshfs-0.0.6.gem
- `env | grep -i vagrant` returns
```
PWD=/Users/dmitry/.vagrant.d/gems/gems
OLDPWD=/Users/dmitry/.vagrant.d/gems/gems/vagrant-digitalocean-0.7.3
```

**Comment by sethvargo** [comment]
Hi @dmatora 

When you say "successfully installed", what exactly do you mean? Are you getting an error or just not seeing them in the gems folder? I'm trying to understand the exact issue here, because I do not see any error messages in your output :smile:

**Comment by dmatora** [comment]
successful install means both folder and files are in place
didn't - means folder is there but it's empty
both cases it produces no errors, except when you try to run a project that depends on a plugin :)

**Comment by sethvargo** [comment]
And what error are you seeing when you run a project that depends on that plugin?

This definitely seems like an issue with those individual plugins. It's hard for me to understand why some of the plugins would install and some would not, especially if you are not getting an error. Can I see the error when the plugin tries to activate?

**Comment by dmatora** [comment]
same error that you get when you don't have plugin
like for nfs_guest it says
```
Vagrant:
* Unknown configuration section 'nfs_guest'.
```

**Comment by sethvargo** [comment]
This is weird... I'm going to tag this as a bug, but I am unable to reproduce this issue (all plugins installed correctly for me on master build and 1.7.2 (latest)).

**Comment by dmatora** [comment]
I can let you play with it on my machine via teamviewer

вторник, 31 марта 2015 г. пользователь Seth Vargo написал:

**Comment by sethvargo** [comment]
Hi @dmatora 

I have tried to reproduce this, and I am simply unable to. I have tried with multiple different plugins, different types of gems, and even on Windows, and I cannot reproduce this.

Vagrant is a fairly large project with a popular plugin ecosystem, and if this was truly a bug in Vagrant, I would expect to see many users chiming in with a +1 or a "me too". Given that, combined with the fact that I am unable to reproduce this, I believe this is an issue that is unique to your environment.

I would recommend removing everything in `~/.vagrant.d` and re-installing the latest version of Vagrant (1.7.4 at this time, 1.8 at the end of the month). Thanks, and sorry! :smile:
